### ISSUE TITLE
Debian 11 will not show any stats

### ISSUE THREAD

**Issue by ranrinc** [ualex73/monitor_docker#67]
After the upgrade, my os from Debian 10 to Debian 11 all my stats disappear from both portainer and monitor_docker. Is there a solution for this? Especially since HA will soon be upgrading its supervised version to Debian 11. 

Debian 10
![Screen Shot 2021-08-28 at 11 23 10 AM](https://user-images.githubusercontent.com/15801696/131218225-42a33357-73f7-4c7b-95ed-f47cb36cc56d.png)

Debian 11
![Screen Shot 2021-08-28 at 11 11 41 AM](https://user-images.githubusercontent.com/15801696/131218227-ca82dd13-e24f-466b-bab6-1c77d012861e.png)

**Comment by ranrinc** [comment]
Just found out that the problem is being Debian 11 is implementing a new cgroup from v1 to v2 which creates changes in getting stats. Will the add-on fixed the problem too?

**Comment by KalleDK** [comment]
I had same problems, as a tmp quick fix I simple changed the following in helpers.py

```yaml
        # Gather memory information
        memory_stats = {}
        try:
            # Memory is in Bytes, convert to MBytes
            memory_stats["usage"] = toMB(
                # raw["memory_stats"]["usage"] - raw["memory_stats"]["stats"]["cache"]
                raw["memory_stats"]["usage"]
            )
            memory_stats["limit"] = toMB(raw["memory_stats"]["limit"])
            # memory_stats["max_usage"] = toMB(raw["memory_stats"]["max_usage"])
            memory_stats["usage_percent"] = round(
                float(memory_stats["usage"]) / float(memory_stats["limit"]) * 100.0,
                PRECISION,
            )
```

**Comment by KalleDK** [comment]
From https://docs.docker.com/engine/reference/commandline/stats/

> On Linux, the Docker CLI reports memory usage by subtracting cache usage from the total memory usage. The API does not perform such a calculation but rather provides the total memory usage and the amount from the cache so that clients can use the data as needed. The cache usage is defined as the value of total_inactive_file field in the memory.stat file on cgroup v1 hosts.

> On Docker 19.03 and older, the cache usage was defined as the value of cache field. On cgroup v2 hosts, the cache usage is defined as the value of inactive_file field.

**Comment by KalleDK** [comment]
Also https://docs.docker.com/engine/api/v1.41/#operation/ContainerStats

**Comment by ranrinc** [comment]
> ```yaml
> raw["memory_stats"]["usage"] - raw["memory_stats"]["stats"]["inactive_file"]
> ```

yay it works.. thanks, man.

**Comment by nickrout** [comment]
How about a PR then?

**Comment by ranrinc** [comment]
> How about a PR then?

The credit goes to @KalleDK ... else I already do a PR hahahaha...

**Comment by KalleDK** [comment]
I don't know how to test if their are cgroup1 or 2 which would be needed in order not to break the older

**Comment by ranrinc** [comment]
> I don't know how to test if there are cgroup1 or 2 which would be needed in order not to break the older

you are correct on that issue. I know how to check using docker command, but that is as far as getting info

`docker info | grep -i cgroup`

**Comment by ualex73** [comment]
Just applied the workaround was mentioned above in the master. Can anybody test the new "helpers.py" out and let me know?

File modified:
https://github.com/ualex73/monitor_docker/blob/master/custom_components/monitor_docker/helpers.py

**Comment by ranrinc** [comment]
> @KalleDK thanks it is merged. I will do a test tomorrow/friday and then release it as a version (then it shows up in HACS)

great.. can wait till it's up. Since the latest HA already making Debian 11 official
